lib/mono/cli/publish.rb

```diff
@@ -161,11 +194,23 @@ def commit_changes(packages)
             ["Publish package #{only_package.next_tag}", standard_message]
           end
         run_command "git add -A"
+        rollback << [
+          "## Removing release commit",
+          "git restore --staged :/"
+        ]
         run_command "git commit -m '#{commit_subject}' -m '#{commit_message}'"
+        rollback << [
+          "## Removing release commit",
+          "git reset --soft HEAD^"
```

Comment:
Do we want to keep the changes uncommitted with `--soft`?

Reply:
What I was going for was, for each of the rollback command entries, to only roll back what was done by the command that immediately precedes it. So the rollback entry added after `git commit` (which takes the staged files and creates a new commit from them) does `git reset --soft HEAD^` (which undoes the latest commit but leaves the files in it staged)

Similarly, right above this one, after running `git add -A` (which stages all files) we add a rollback entry for `git restore --staged :/` (which unstages all files)

And after modifying the files (to update the packages and the changelog) we add rollback entries for `git restore :/` (which undoes the updates)

So the end result of _the whole rollback_ should be that the changes are undone entirely, not left uncommitted.

Reply:
okay sounds good 👍